packages/@tinacms/cli/src/next/commands/build-command/index.ts

```diff
@@ -598,6 +608,79 @@ export class BuildCommand extends BaseCommand {
       }
     }
   }
+
+  async checkTinaSchema(
+    configManager: ConfigManager,
+    database: Database,
+    apiURL: string,
+    previewName: string,
+    verbose: boolean
+  ) {
+    const bar = new Progress(
+      'Checking local Tina Schema matches server. :prog',
+      1
+    )
+    const { config } = configManager
+    const token = config.token
+    const { clientId, branch, isLocalClient, host } = parseURL(apiURL)
+    // Can't check status if we're not using Tina Cloud
+    if (isLocalClient || !host || !clientId || !branch) {
+      if (verbose) {
+        logger.info(logText('Not using Tina Cloud, skipping Tina Schema check'))
+      }
+      return
+    }
+
+    // Get the remote schema from the graphql endpoint
+    const { tinaSchema: remoteTinaSchemaSha } = await fetchSchemaSha({
+      url: `https://${host}/db/${clientId}/${previewName || branch}/schemaSha`,
+      token,
+    })
+
+    if (!remoteTinaSchemaSha) {
+      bar.tick({
+        prog: '❌',
+      })
+      let errorMessage = `The remote Tina schema does not exist. Check indexing for this branch.`
+      if (config?.branch) {
+        errorMessage += `\n\nAdditional info: Branch: ${config.branch}, Client ID: ${config.clientId} `
+      }
+      throw new Error(errorMessage)
+    }
+
+    if (!database.bridge) {
+      throw new Error(`No bridge configured`)
+    }
+    try {
+      const localTinaSchema = JSON.parse(
+        await database.bridge.get(
+          path.join(database.tinaDirectory, '__generated__', '_schema.json')
+        )
+      )
+      localTinaSchema.version = undefined
+      const localTinaSchemaSha = crypto
+        .createHash('sha256')
+        .update(JSON.stringify(localTinaSchema))
+        .digest('hex')
+
+      if (localTinaSchemaSha === remoteTinaSchemaSha) {
+        bar.tick({
+          prog: '✅',
+        })
+      } else {
+        bar.tick({
+          prog: '❌',
+        })
+        let errorMessage = `The local Tina schema doesn't match the remote Tina schema. Please push up your changes to GitHub to update your remote tina schema.`
+        if (config?.branch) {
+          errorMessage += `\n\nAdditional info: Branch: ${config.branch}, Client ID: ${config.clientId} `
+        }
+        throw new Error(errorMessage)
+      }
+    } catch (e) {
+      throw e
```

Comment:
**question**: Why catch and rethrow the error?

Reply:
No reason - it was just leftover from copy/paste of existing function